src/elastio-snap.c

```diff
@@ -3344,6 +3346,15 @@ static void __on_bio_read_complete(struct bio *bio, int err){
 	bio_free_clone(bio);
 }
 
+static inline void wait_for_bio_complete(struct snap_device *dev)
+{
+	struct bio_queue *bq = &dev->sd_cow_bios;
+	if (atomic64_read(&dev->sd_received_cnt) != atomic64_read(&dev->sd_processed_cnt))
+		wait_event_interruptible_timeout(bq->event,
+				atomic64_read(&dev->sd_submitted_cnt) == atomic64_read(&dev->sd_processed_cnt),
+				msecs_to_jiffies(500));
```

Comment:
The conditions in `if` and in the `wait_event_interruptible_timeout` are different. Discussed it with @skypodolsky offline. The code will be changed.

Reply:
Exactly, this has been fixed.